server/controllers/comment.js

```diff
@@ -0,0 +1,104 @@
+const { GeneralError, NotFound, Unauthorized } = require('../utils/errors');
+const { validationResult } = require('express-validator');
+
+const Profile = require('../models/profile');
+const Post = require('../models/post');
+const Comment = require('../models/Comment');
+
+exports.addComment = async (req, res, next) => {
+    const errors = validationResult(req);
+
+    if (!errors.isEmpty())
+        return res.status(400).json({
+            error: errors.array(),
+        });
+
+    const { postId } = req.params;
+    const { userId, text } = req.body;
+
+    try {
+        const post = await Post.findById(postId);
+        if (!post) throw new NotFound('No post found');
+
+        const profile = await Profile.findOne({ user: userId });
+        if (!profile) throw new NotFound('No profile found');
+
+        const comment = new Comment({
+            user: userId,
+            text,
+            name: `${profile.firstName} ${profile.lastName}`,
+            avatar: profile.imageUrl,
+        });
+
+        const commentRes = comment.save();
+        if (!commentRes) throw new GeneralError('Error adding a comment');
+
+        post.comments.unshift(comment);
```

Comment:
I notice from this that your plan for the comments is:
[`most recent comment`, `comment`, `comment`....., `first. comment`]

Adding an element to the beginning of an array (unshift) is a much less efficient task than pushing an element to the end of an array.  This is because every single element in the array needs to have its reference updated (element at index 0 is now 1, 1 is 2, 2 is 3, so on).  If the array length increases, this becomes an increasingly taxing operation to perform.

In comparsion, Array.push simply appends a new element to the end of the array (with a new index of `length`) and then adds 1 to `Array.length`.  Always efficient, no matter the length of the array.  I'd recommend switching to that instead, so:

[`first comment`, `second comment`....., `latest comment`]

Reply:
Changed unshift method to push. It was an interesting thing to know.